### PR TITLE
Add Automitigator that can identify mitigations by searching.

### PR DESCRIPTION
**Pull request by yrajas**
* Uses FLAML to define a search space of all possible mitigations.
* Searches mitigation configurations from the search space and applies those mitigations to the dataset.
* Evaluates different configurations in accordance with the time/number of searches constraints to identify the best sequence of mitigations and AutoML model.

### CODE REVIEW

File: raimitigations/automitigator/evaluator.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        #            'mitigations':
+        #               {'action0': {'type': 0, 'strategy': 0, 'name': 'rebalancer'}}}}
+
+        self.pipeline = None
+        search_space = search_config["search_space"]
+        cohort = search_space["cohort"]
+        if cohort == "all":
+            return self.mitigate_full_dataset(train_x, train_y, search_space)
+        else:
+            raise ValueError(f"Unknown cohort type {cohort}")
```
Comment (mrfmendonca) [inline raimitigations/automitigator/evaluator.py:56]: Does it support only ne type of cohort for now (``cohort = 'all'``)?

File: raimitigations/automitigator/evaluator.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    def _process_feature_selector(self, selector_type):
+        """
+        Process the feature selector
+
+        :param dict selector_type: The feature selector configuration
+
+        :raises ValueError: If the feature selector is unknown
+        """
+        selector_name = selector_type["selector_name"]
```
Comment (mrfmendonca) [inline raimitigations/automitigator/evaluator.py:66]: Usually, I like to define these strings in some sort of class with all relevant constants. This way, if we ever need to change one of these values, we can do that easily by changing only one value. For example, if you decide to change from '``selector_name``' to '``name_selector``', you'll need to do a search and replace for all '``selector_name``' occurrences. But if you have all constants in a single place, you only need to change it there.

File: raimitigations/automitigator/evaluator.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                )
+            )
+
+    def mitigate_full_dataset(self, train_x, train_y, search_space):
+        """
+        Evaluates a given set of mitigations on a given dataset.
+
+        :param train_x: The training data
+        :param train_y: The training labels
+        :param dict search_space: The search configuration
```
Comment (mrfmendonca) [inline raimitigations/automitigator/evaluator.py:100]: This ``search_space`` parameter needs a more in-depth documentation. I more detailed docstring for it, or some examples of different use cases in a notebook inside the ``notebooks/`` folder would help a lot.

**Review by mrfmendonca — APPROVED** [review]

File: raimitigations/automitigator/evaluator.py (hunk 2 above)
Comment (yrajas) [reply raimitigations/automitigator/evaluator.py:66]: Fixed.

File: raimitigations/automitigator/evaluator.py (hunk 1 above)
Comment (yrajas) [reply raimitigations/automitigator/evaluator.py:56]: Yes, for now. 
The cohort specific mitigations is the next step to be addressed.

File: raimitigations/automitigator/evaluator.py (hunk 3 above)
Comment (yrajas) [reply raimitigations/automitigator/evaluator.py:100]: Agreed. A separate notebook with usage information is planned for separately.

File: raimitigations/automitigator/__init__.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+from .automitigator import AutoMitigator
```
[the remark this answers is not in the document]
Comment (yrajas) [reply raimitigations/automitigator/__init__.py:1]: Absolutely. All documentation changes are going to be in a separate PR. Will be merging the code changes for now.

**Pull request merged**